content/docs/rfcs/34/README.md

```diff
@@ -0,0 +1,123 @@
+---
+slug: 34
+title: 34/WAKU2-PEER-EXCHANGE
+name: Waku v2 Peer Exchange
+status: raw
+category: Standards Track
+tags: waku/core-protocol
+editor: Daniel Kaiser <danielkaiser@status.im>
+contributors:
+---
+
+# Abstract
+
+This document specifies a simple request-response peer exchange protocol.
+Responders send information about a requested number of peers.
+The main purpose of this protocol is providing resource restricted devices with peers.
+
+# Background and Motivation
+
+Resource restricted devices may not feasibly be able to take part in distributed random sampling ambient peer discovery protocols such as [33/WAKU2-DISCV5](https://rfc.vac.dev/spec/33/).
+The Waku peer discovery protocol specified in this document allows resource restricted devices to request a list of peers from a (stronger) peer.
+Network parameters necessary to connect to this stronger peer can, for instance, be learned from a static bootstrapping method.
+The advantage of using Waku peer exchange to discover new peers over simply adding static bootstrap peers to the peer list is a more even load distribution.
+If a lot of (resource restricted) nodes would use static bootstrap nodes as relay or store nodes, the load on these would be very high.
+Heavily used static nodes also add a centralized element. Downtime of such a node might significantly impact the network.
+
+However, the resource efficiency of this protocol comes at an anonymity cost, which is explained in the [Security/Privacy Considerations](#securityprivacy-considerations) section.
+This protocol SHOULD only be used if [33/WAKU2-DISCV5](https://rfc.vac.dev/spec/33/) is infeasible.
+
+# Theory and Protocol Semantics
+
+The peer exchange protocol specified in this document is a simple request-response protocol.
+As Figure 1 illustrates, the requesting node sends a request to a peer, which acts as the responder.
+The responder replies with a list of [multiaddresses](https://docs.libp2p.io/concepts/addressing/).
+
+![Figure 1: The responder provides a list of multiaddresses of peers to the requester.](../../../../static/rfcs/34/protocol.svg)
+
+In order to protect its anonymity, the responder MUST NOT provide peers from its actively used peer list as this opens pathways to *Neighbourhood Surveillance* attacks, as described in the
+[Security/Privacy Considerations Section](#securityprivacy-considerations).
+The responder SHOULD retrieve a new random set of peers via [33/WAKU2-DISCV5](https://rfc.vac.dev/spec/33/).
```

Comment:
At the moment, I wanted to leave that open.
I should discuss this in the "implantation suggestions" section.

Its a trade-off between anonymity and efficiency.
Sending the same set of peers to various nodes opens pathways to neighbourhood surveillance.

Reply:
For efficiency purposes, especially in the context of a web app loading, it would be better if the responder retrieves and cache a random set of peers via discv5 first and then deliver some of those peers in the response I suggest we strongly recommended the .

If the responder triggers discv5 upon receiving a request, I am afraid the latency would be too high to be usable in a web app context.

I can see this is also mentioned in the Security section.

I think we should rephrase this to discourage discv5 being trigger upon request and encourage the usage of a cache.

Reply:
Addressed in c02c27ee64 while protecting the anonymity properties. I did not thoroughly analyse the implications of reusing peers in response sets yet. But the phrasing now makes caching a SHOULD and reusing exchange peers a MAY.